### PR TITLE
HAUKI-289 Render state period status on resource page

### PR DESCRIPTION
**Pull request by VilleMiekkoja**
image

### CODE REVIEW

File: src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx [hunk 1]
```diff
@@ -26,6 +26,38 @@ function findResourceStateLabelByValueAndLang(
   return resourceStateObj ? resourceStateObj[language] : null;
 }
 
+function renderPeriodStatusByLanguage(
+  periodStatus: ResourceState | undefined,
+  language: Language,
+  stateOptions: UiOptionsFieldConfig
+): JSX.Element {
+  if (periodStatus === 'open' || periodStatus === 'closed') {
```
Comment (VilleMiekkoja) [inline src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx:34]: Im not sure if open and closed requires this prefix 'always'? What do you think? Should we extend the prefix to all status or remove it completely? Or leave it as it is? https://helsinkisolutionoffice.atlassian.net/browse/HAUKI-289

File: src/resource/resource-opening-hours/opening-period/OpeningPeriodDetails.tsx [hunk 2]
```diff
@@ -30,8 +30,12 @@ function containsSpecialRules(rules: GroupRule[]): boolean {
 function getNonSupportedFeatures(datePeriod: DatePeriod): string[] {
   const nonSupportedFeatures = [];
   if (datePeriod.time_span_groups.length === 0) {
-    nonSupportedFeatures.push('Jakso ilman aukioloryhmää');
-    return nonSupportedFeatures;
+    if (datePeriod.resource_state === 'undefined') {
```
Comment (minevala) [inline src/resource/resource-opening-hours/opening-period/OpeningPeriodDetails.tsx:33]: Enum can be used in comparison:
```js
if (datePeriod.resource_state === ResourceState.UNDEFINED) {
```

File: src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx (hunk 1 above)
Comment (minevala) [reply src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx:34]: I'm not sure do these even need prefix at all 🤔
We could leave it as it is and ask for more opinions.

Enums can be used in comparisons:
```js
 if (
    periodStatus === ResourceState.OPEN ||
    periodStatus === ResourceState.CLOSED
  ) {
```

**Review by minevala — COMMENTED** [review]
Works 👍 Few minor comments

**Review by Rikuoja — CHANGES_REQUESTED** [review]
Looks good! 👍  But yeah perhaps only render the text "Suljettu" without "aina".

File: src/resource/resource-opening-hours/opening-period/OpeningPeriodDetails.tsx (hunk 2 above)
Comment (VilleMiekkoja) [reply src/resource/resource-opening-hours/opening-period/OpeningPeriodDetails.tsx:33]: - [ ] Apply Enum here

File: src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx (hunk 1 above)
Comment (VilleMiekkoja) [reply src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx]: - [ ] Remove prefixes

File: src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx [hunk 3]
```diff
@@ -87,39 +101,51 @@ export default function PeriodOpeningHours({
...
-        );
-      })}
+      {datePeriodStatus !== 'undefined' &&
+        renderPeriodStatusByLanguage(
+          datePeriodStatus,
+          language,
+          datePeriodConfig.resourceState
+        )}
+      {datePeriodStatus === 'undefined' &&
+        timeSpans.map((timeSpan, index) => {
```
Comment (minevala) [inline src/resource/resource-opening-hours/opening-period/PeriodOpeningHours.tsx:117]:
```suggestion
      {datePeriodStatus !== ResourceState.UNDEFINED ?
       renderPeriodStatusByLanguage(
          datePeriodStatus,
          language,
          datePeriodConfig.resourceState
        )  :   
        timeSpans.map((timeSpan, index) => {
```

**Review by minevala — APPROVED** [review]
👍

**Pull request merged**